avro/swagger/src/main/java/com/cerner/beadledom/avro/SwaggerAvroModelConverter.java

```diff
@@ -119,49 +116,39 @@ protected String getName(Schema schema) {
    * determined merely by looking at the schema. It may also be used recursively to build collection
    * and union types.
    *
-   * @return the parsed property, or null if it cannot/should not be represented in the Swagger
-   *     model
+   * @return the parsed property, or null if it cannot/should not be represented in the Swagger model
    */
   @Nullable
-  protected ModelProperty parseSchema(Schema schema) {
+  protected Property parseSchema(Schema schema) {
     switch (schema.getType()) {
       case RECORD:
-        return simpleProperty(getName(schema), schema.getFullName());
+        // todo PropertyBuilder.build() returning null
+        return null;
+//        Property record = PropertyBuilder.build(getName(schema), schema.getFullName(), Collections.emptyMap());
+//        record.setRequired(true);
+//        return record;
       case ENUM:
         // TODO may wish to include the enum's documentation in the field documentation, since it
         // won't appear anywhere else
-        return new ModelProperty(
-            "string",
-            "string",
-            0,
-            true,
-            Option.<String>empty(),
-            new AllowableListValues(
-                JavaConversions.asScalaBuffer(schema.getEnumSymbols()).toList(),
-                "LIST"),
-            Option.<ModelRef>empty()
-        );
+        StringProperty property = new StringProperty();
+        property.required(true);
+        property.setEnum(schema.getEnumSymbols());
+
+        return property;
       case ARRAY:
-        ModelProperty elementsProperty = parseSchema(schema.getElementType());
+        Property elementsProperty = parseSchema(schema.getElementType());
         if (elementsProperty == null) {
           return null;
         }
 
-        if (SwaggerSpec.containerTypes().contains(elementsProperty.type())) {
-          LOGGER.debug("Cannot include nested collection schema in swagger docs: {}", schema);
-          return null;
-        }
-        return new ModelProperty(
-            "List",
-            "List",
-            elementsProperty.position(),
-            true,
-            elementsProperty.description(),
-            elementsProperty.allowableValues(),
-            Option.apply(modelRef(elementsProperty))
-        );
+        Property arrayProperty = new ArrayProperty(elementsProperty);
+        arrayProperty.setRequired(true);
+
+        return arrayProperty;
       case BOOLEAN:
-        return simpleProperty("boolean", "boolean");
+        Property booleanProperty = new BooleanProperty();
+        booleanProperty.setRequired(true);
```

Comment:
Are boolean properties always going to be required?

Reply:
Everything that is not a union with null is required.